### PR TITLE
change route and fix little bug

### PR DESCRIPTION
**Pull request by herve-perel**
## Description

Please include a summary of the change or which issue is fixed.

## Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Area of change

- [x] Frontend
- [x] Backend
- [ ] Other: .....

## General checklist:

- [x] My code follows the style guidelines of this project
- [x] I ran `bun run format`/`bun run lint` before commit
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation where needed
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [ ] I have updated `develop` and merged to my branch before submitting pull request
- [x] My pull request generate no conflicts with `develop` branch
- [ ] I requested code review from other team members

## Frontend checklist:

- [x] I followed guidelines for `HTML`/`JSX`/`Typescript`
- [x] My `Javascript` generate no new console errors
- [x] I tested my code cross browsers
- [x] My slice is pixel perfect for both desktop and mobile according to design
- [ ] I conducted basic QA to assure all features are working
- [x] I tested responsive for mobile and tablet resolutions

## Backend checklist:

- [ ] I tested admin by manually adding content from zero
- [ ] I created easy to use admin experience which is self-explanatory
- [ ] I added description to admin fields in hard-to-understand areas
- [ ] I followed guidelines for naming `Typescript` variables
- [ ] I conducted basic QA to assure all features are working

### CODE REVIEW

**Review by Abecker37 — DISMISSED** [review]

**Review by JorisPineau — APPROVED** [review]

File: packages/backend/api/src/my-informations.ts [hunk 1]
```diff
@@ -1,28 +1,39 @@
...
+const infoRouter = express.Router();
 
-    if (user.length === 0) {
-      return res.status(404).json({ error: 'User not found' });
+infoRouter.get(
+  '/',
+  loginIdUser,
+  async (req: RequestWithUser, res: Response) => {
+    if (!req.userId) {
+      throw new Error('User is not authenticated');
```
Comment (JorisPineau) [inline packages/backend/api/src/my-informations.ts:18]: catch

**Review by JorisPineau — DISMISSED** [review]

File: packages/backend/api/src/my-preferences.ts [hunk 2]
```diff
@@ -1,71 +1,88 @@
...
 
-    await db
-      .updateTable('user')
-      .set({
+preferenceRouter.put(
+  '/',
+  loginIdUser,
+  async (req: RequestWithUser, res: Response) => {
+    if (!req.userId) {
+      throw new Error('User is not authenticated');
```
Comment (JorisPineau) [inline packages/backend/api/src/my-preferences.ts:53]: LA

**Review by JorisPineau — DISMISSED** [review]

File: packages/backend/api/package.json [hunk 3]
```diff
@@ -19,7 +19,6 @@
   },
   "devDependencies": {
     "@app/types": "workspace:*",
-    "@types/multer": "^1.4.11",
```
Comment (YDrogen) [inline packages/backend/api/package.json:22]: Pourquoi retirer le type ? Tu as supprimé Multer ?

File: packages/frontend/web/src/pages/MyPreferences.tsx [hunk 4]
```diff
@@ -13,6 +13,34 @@ import {
...
+        is_non_vaccinated_allowed: userData.is_non_vaccinated_allowed,
+        is_animal_allowed: userData.is_animal_allowed,
+        is_smoker_allowed: userData.is_smoker_allowed,
+        selected_languages: JSON.parse(userData.selected_languages),
+        selected_musics: JSON.parse(userData.selected_musics),
+      }),
+    });
+
+    if (response.ok) {
+      ('Data saved successfully');
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/MyPreferences.tsx:35]: Que fait cette ligne ?

**Review by YDrogen — CHANGES_REQUESTED** [review]

File: packages/backend/api/package.json (hunk 3 above)
Comment (herve-perel) [reply packages/backend/api/package.json:22]: il y avait deux fois la ligne multer

**Review by Abecker37 — APPROVED** [review]

**Pull request merged**